backend/app/adapter/sqldb/short_link.go

```diff
@@ -286,6 +286,33 @@ func (s ShortLinkSQL) composeParamList(numParams int) string {
 	return parameterStr
 }
 
+// DeleteShortLink deletes an existing user short link.
+func (s ShortLinkSQL) DeleteShortLink(shortLinkInput entity.ShortLinkInput) error {
+	query := fmt.Sprintf(`
+	DELETE FROM %s
+	WHERE %s=$1
+	`,
+		table.ShortLink.TableName,
+		table.ShortLink.ColumnAlias,
+	)
+
+	alias := shortLinkInput.GetCustomAlias("")
+	result, err := s.db.Exec(query, alias)
+	if err != nil {
+		return err
+	}
+
+	affectedRowCount, err := result.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if affectedRowCount == 0 {
+		return fmt.Errorf("failed to delete user short link alias(%s)", alias)
```

Comment:
We can make this a reusable error inside repository so that all repos can leverage it! https://github.com/short-d/short/tree/master/backend/app/usecase/repository